Homework 8

Dragos-Andrei-75 opened the pull request:
Setup One to Many and Many to One relations between Exemplary (owner) and Book entities.

Defined endpoints:
- createMultiple
    - handles requests to create multiple Exemplary entities;
- listPaginated
    - handles requests to list the Exemplary entities on a page;
- delete
    - handles requests to delete an Exemplary entity;

Defined classes:
- Exemplary (all packages);
- ExemplarsCreateDTO (entitiesDTO);
- ExemplarsResponseDTO (entitiesDTO);

Dragos-Andrei-75 committed:
Homework 8

Setup One to Many and Many to One relations between Exemplary (owner) and Book entities.

Defined endpoints:
- createMultiple
    - handles requests to create multiple
      Exemplary entities;
- listPaginated
    - handles requests to list the
      Exemplary entities on a page;
- delete
    - handles requests to delete an
      Exemplary entity;

Defined classes:
- Exemplary (all packages);
- ExemplarsCreateDTO (entitiesDTO);
- ExemplarsResponseDTO (entitiesDTO);

marcopacurariu3 commented on src/main/java/com/example/SpringBookstore/controller/ExemplaryController.java:
```diff
@@ -0,0 +1,44 @@
+    private ExemplaryService exemplaryService;
+
+    @PostMapping
+    public ResponseEntity<?> createMultiple(@RequestBody ExemplarsCreateDTO exemplarsCreateDTO) {
```
just "create"

marcopacurariu3 commented on src/main/java/com/example/SpringBookstore/mapper/ExemplaryMapper.java:
```diff
@@ -0,0 +1,64 @@
+
+@Component
+public class ExemplaryMapper {
+    private static BookRepository bookRepository;
```
static? 🤔

marcopacurariu3 commented on src/main/java/com/example/SpringBookstore/mapper/ExemplaryMapper.java:
```diff
@@ -0,0 +1,64 @@
+        ExemplaryMapper.bookRepository = bookRepository;
+    }
+
+    public static Exemplary exemplaryDTO2Exemplary(ExemplaryDTO exemplaryDTO) {
```
il apelezi pe undeva? sau l-ai facut doar asa? ca mi-a sarit in ochi ca din DTO, cand faci entitatea, ii pui si cartea, ceea ce nu cred ca e ok pentru ca nu creezi un exemplar, impreuna cu o carte. cel mult o sa cream o carte impreuna cu exemplare. si atunci nu o sa ai un caz in care primesti de pe front-end, pentru un exemplar, ca sa il creezi, mai mult decat un bookId. deci nu vei primi toate informatiile despre carte impreuna cu exemplarul

marcopacurariu3 commented on src/main/java/com/example/SpringBookstore/mapper/ExemplaryMapper.java:
```diff
@@ -0,0 +1,64 @@
+        return exemplaryDTO;
+    }
+
+    public static List<Exemplary> exemplarsCreateDTO2Exemplars(ExemplarsCreateDTO exemplarsCreateDTO) {
```
not really, in mapper nu ar trebui sa accesam repository-ul direct. Ideea e ca semnatura metodei e excelenta. asa trebuie sa faci, dar doar creezi o lista de exemplare, cu care ulterior apelezi un service (impreuna cu bookId, da). Si acolo faci logica de genul asta. Astfel nu mergi nici cu CreateDTO-ul pana la Service.

marcopacurariu3 replied:
Mapperele doar ar trebui sa converteasca date dintr-un format in altul ideal, fara sa fie nevoie sa apeleze service-uri & stuff. Cel putin la o aplicatie cu aceste layere cum facem noi, cand ajungem la cazuri mai complexe, mai vedem

marcopacurariu3 commented on src/main/java/com/example/SpringBookstore/service/ExemplaryService.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    public void delete(Long exemplaryID) {
+        if (exemplaryRepository.existsById(exemplaryID)) exemplaryRepository.deleteById(exemplaryID);
```
sa pui si acolade, pls

marcopacurariu3 commented on src/main/java/com/example/SpringBookstore/service/ExemplaryService.java:
```diff
@@ -0,0 +1,43 @@
+    @Autowired
+    private ExemplaryRepository exemplaryRepository;
+
+    public ExemplarsResponseDTO createMultiple(ExemplarsCreateDTO exemplarsCreateDTO) {
```
si aici, returnam o lista de exemplare (cele pe care am dat save())

Dragos-Andrei-75 committed:
Addressed All Issues from the Previous Review

Addressed all issues from the previous review made by Marco Păcurariu.

Added the Bookstore.postman_collection.json file into the resources folder.

Bookstore.postman_collection.json contains request for testing each endpoint that exist within the project.
